pkg/profiler/cpu/cpu.go

```diff
@@ -263,54 +271,40 @@ func (p *CPU) Run(ctx context.Context) error {
 		return fmt.Errorf("failure updating: %w", err)
 	}
 
-	stackCounts, err := m.GetMap(stackCountsMapName)
+	p.bpfMaps, err = initializeMaps(m, p.byteOrder)
 	if err != nil {
-		return fmt.Errorf("get counts map: %w", err)
+		return fmt.Errorf("failed to initialize eBPF maps: %w", err)
 	}
 
-	stackTraces, err := m.GetMap(stackTracesMapName)
-	if err != nil {
-		return fmt.Errorf("get stack traces map: %w", err)
-	}
-	unwindTable1, err := m.GetMap(unwindTable1MapName)
-	if err != nil {
-		return fmt.Errorf("get unwind table 1 map: %w", err)
-	}
-	unwindTable2, err := m.GetMap(unwindTable2MapName)
-	if err != nil {
-		return fmt.Errorf("get unwind table 2 map: %w", err)
-	}
-	unwindTable3, err := m.GetMap(unwindTable3MapName)
-	if err != nil {
-		return fmt.Errorf("get unwind table 3 map: %w", err)
-	}
+	if len(p.debugProcessNames) > 0 {
+		level.Info(p.logger).Log("msg", "process names specified, debugging processes", "metchers", strings.Join(p.debugProcessNames, ", "))
+		pfs, err := procfs.NewDefaultFS()
+		if err != nil {
+			return fmt.Errorf("failed to create procfs: %w", err)
+		}
 
-	dwarfStackTraces, err := m.GetMap(dwarfStackTracesMapName)
-	if err != nil {
-		return fmt.Errorf("get dwarf stack traces map: %w", err)
-	}
+		var matchers []*regexp.Regexp
+		for _, exp := range p.debugProcessNames {
+			regex, err := regexp.Compile(exp)
```

Comment:
compiling a regex is rather expensive, we should do this once only

Reply:
This actually happens once. It's outside the profiling loop and the watch loop.